datafusion/functions/src/regex/regexpcount.rs

```diff
@@ -548,19 +549,25 @@ where
     }
 }
 
-fn compile_and_cache_regex<'a>(
-    regex: &'a str,
-    flags: Option<&'a str>,
-    regex_cache: &'a mut HashMap<String, Regex>,
-) -> Result<&'a Regex, ArrowError> {
-    if !regex_cache.contains_key(regex) {
+fn compile_and_cache_regex<'strings, 'cache>(
+    regex: &'strings str,
+    flags: Option<&'strings str>,
+    regex_cache: &'cache mut HashMap<(&'strings str, Option<&'strings str>), Regex>,
+) -> Result<&'cache Regex, ArrowError>
+where
+    'strings: 'cache,
```

Comment:
I've never seen this before. Does this effectively mean 'strings is equivalent to 'cache ?

Reply:
Yeah, that was I was looking into, I think its saying that strings comes before the cache and lives after as well. I don't know if that is the correct behaviour

Reply:
"strings outlives hashmap" or "strings lasts longer than hashmap", so hashmap can borrow data from strings (because while hashmap holds data from strings, strings won't die), but if hashmap dies before strings dies, it's fine.